tenseal/cpp/tensors/encrypted_tensor.h

```diff
@@ -161,6 +161,10 @@ class EncryptedTensor {
      **/
     void link_tenseal_context(shared_ptr<TenSEALContext> ctx) {
         this->_context = ctx;
+        if (_lazy_buffer) {
+            this->load(*_lazy_buffer);
+            _lazy_buffer = {};
+        }
```

Comment:
maybe have an else here which throw a weird error

Reply:
The "else" case here is the standard call to "link_tenseal_context", used in constructors. we should not throw anything there. We just want to unpack the optional lazy_buffer, otherwise we will rely on the null context exceptions.

Reply:
For example, this line would go in the "else" branch https://github.com/OpenMined/TenSEAL/blob/master/tenseal/cpp/tensors/ckksvector.cpp#L11

Reply:
Got it!